omod-1.9/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs1_9/ConceptController1_8Test.java

```diff
@@ -212,6 +212,27 @@ public void shouldReturnTheAuditInfoForTheFullRepresentation() throws Exception
 		Assert.assertNotNull(result);
 		Assert.assertNotNull(PropertyUtils.getProperty(result, "auditInfo"));
 	}
+
+	@Test
+	public void shouldSearchAndReturnConceptsFilteredByClassAndName() throws Exception {
+		MockHttpServletRequest req = request(RequestMethod.GET, getURI());
+		SimpleObject result;
+		List<Object> hits;
+
+		String conceptClassUuid = "3d065ed4-b0b9-4710-9a17-6d8c4fd259b7"; // DRUG
+		String name = "Aspirin"; //ASPIRIN
+		String searchType = "equals";

```

Comment:
Does this test alone cater for the two if and else paths that you have introduced above?